File: solutions/scala/012-longest-common-prefix.scala
/**
 * Problem 12: Longest Common Prefix (LeetCode 14)
 * Difficulty: Easy
 * Language: Scala
 */
def longestCommonPrefix(strs: Array[String]): String = {
  if (strs.isEmpty) ""
  else strs.reduce { (prefix, s) =>
    prefix.zip(s).takeWhile { case (a, b) => a == b }.map(_._1).mkString
  }
}

File: solutions/rust/012-longest-common-prefix.rs
/**
 * Problem 12: Longest Common Prefix (LeetCode 14)
 * Difficulty: Easy
 * Language: Rust
 */
pub fn longest_common_prefix(strs: Vec<String>) -> String {
    if strs.is_empty() { return String::new(); }
    strs[1..].iter().fold(strs[0].clone(), |prefix, s| {
        prefix
            .chars()
            .zip(s.chars())
            .take_while(|(a, b)| a == b)
            .map(|(a, _)| a)
            .collect()
    })
}
